pkg/controller/hosted/controller.go

```diff
@@ -182,24 +187,57 @@ func klusterletNamespace(managedCluster string) string {
 	return fmt.Sprintf("klusterlet-%s", managedCluster)
 }
 
+// getHostedManifestWorks gets klusterlet and managed kubeconfig manifest works in the management cluster namespace
+func (r *ReconcileHosted) getHostedManifestWorks(ctx context.Context, cluster *clusterv1.ManagedCluster) ([]workv1.ManifestWork, error) {
+	hostedManifestWorks := []workv1.ManifestWork{}
+
+	managementCluster, err := helpers.GetHostingCluster(cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	klusterletManifestWork := &workv1.ManifestWork{}
```

Comment:
The code of getting the klusterlet manifestwork and getting the kubeconfig manifestwork is a little bit repetitive, could you extract the logic to a function?

Reply:
updated